### DISCUSSION TITLE
Shouldn't the Floating-UI library be considered instead of TippyJS?

### DISCUSSION THREAD

**Discussion by GitHubish** [ueberdosis/tiptap#2737] [General Questions] [answered]
[FloatingUI](https://floating-ui.com/) has been completely rewritten in Typescript and is very modular, powerful and lighter than TippyJS with full documentation. So many reasons that it seems to me that it would make sense to replace TippyJS. What do you think about it?

[any plan to refactor with Floating UI?](https://github.com/atomiks/tippyjs/discussions/1035)

**Comment by bdbch** [comment]
For now we don't really considered changing the popover/popup solution but everything is open for discussion. Are there more information on how lighter it is or any other direct differences between Tippy and FloatingUI?

**Reply by SalahAdDin** [reply]
How ended this discussion?

**Reply by bdbch** [reply]
FloatingUI is implemented in 3.0.0 going forward

**Comment by websiddu** [comment]
Bumping the discussion up, has no updates since Feb 2022 https://github.com/atomiks/tippyjs
While floating ui is constantly developing — https://github.com/floating-ui/floating-ui 

floating ui is used in popular frameworks such as react radix and vue-raidx, currently if you are using tiptap and a one of those libraries tippy is adding significant size along with floating ui library. 

Please do consider converting to floating UI @bdbch is you are open to it I happy to start working on a PR.

**Reply by ajayvignesh01** [reply]
+1

Even the tippy.js maintainers advise to use Floating UI

**Reply by james-pratama** [reply]
+1

**Reply by nperez0111** [reply]
I'd be happy to accept a PR on this, if you are still willing @websiddu

**Reply by SalahAdDin** [reply]
@websiddu will you?

**Comment by bdbch** [comment, marked as answer]
Tippy was deprecated from Tiptap and was replaced by FloatingUI with 3.0.0

**Reply by tjbck** [reply]
We're experiencing a lot of styling issues with v3.0 (FloatingUI) unfortunately so we're currently staying on v2 for Floating & Bubble menus due to this, would it be possible to still include tippy.js as a fallback option?

**Reply by bdbch** [reply]
We don't want to support Tippy.js anymore. What issues are you running into styling related?

**Reply by tjbck** [reply]
Hi, thanks so much for the quick response and for all the hard work that's gone into Tiptap 3.0, it's really appreciated.

To give some specifics: most of our issues are related to z-index handling. With tippy.js, popups consistently appear at the very front layer, regardless of the app context. With FloatingUI, we're running into two problems:

- Initialization: The popup is displayed by default when it should be hidden by default. This could definitely be a compatibility issue on our end (we're using Svelte/Vanilla JS), but it's been tricky to debug.
- Layering & Placement: The popup isn't being reliably positioned above everything else, z-index seems to have less predictable behavior, and sometimes the popup gets clipped or hidden behind other elements. It also seems to extend the container div, whereas tippy.js rendered the popup outside the flow so it could "break free" from any stacking contexts.

If you have any recommendations for properly handling FloatingUI’s layering or best practices for initialization/visibility (especially in non-React setups), that would be super helpful!

Thanks again for your time and all the work you're doing to improve the library.

**Reply by tjbck** [reply]
Screenshot by @Classic298 

<img width="1131" height="880" alt="image" src="https://github.com/user-attachments/assets/859e7ecb-d4a0-44dc-9453-614557ec94a7" />

**Reply by bdbch** [reply]
That's weird, the menu should not be visible from the get-go but should just be mounted if `shouldShow` returns a truthy value.

What does your Menu implementation look like?

**Reply by tjbck** [reply]
@bdbch Using with vanilla js, is there a recommended reference implementation?
